File: iosApp/iosApp/Feature/Menu/Component/AdditionalView.swift
//
//  AdditionalView.swift
//  iosApp
//
//  Created by Егор Лядский on 2023-09-17.
//  Copyright © 2023 orgName. All rights reserved.
//

import SwiftUI
import shared

struct AdditionalInfo: Identifiable {
    
    let id: Int
    let title: String
    let onClick: () -> Void
}

struct AdditionalView: View {
    
    let viewModelWrapper: StatefulViewModelWrapper<MenuViewModelImpl, MenuState>
    let action: () -> Void
    
    var body: some View {
        
        let additionalList = [
            AdditionalInfo(id: 1, title: "О нас", onClick: {
                viewModelWrapper.viewModel.navigateToAboutFairLess()
                action()
            }),
            AdditionalInfo(id: 2, title: "Правила поведения", onClick: {
                viewModelWrapper.viewModel.navigateToRules()
                action()
            }),
            AdditionalInfo(id: 3, title: "Обратная связь", onClick: {
                viewModelWrapper.viewModel.navigateToFeedback()
                action()
            }),
            AdditionalInfo(id: 4, title: "FAQ", onClick: {
                viewModelWrapper.viewModel.navigateToFaq()
                action()
            }),
            AdditionalInfo(id: 5, title: "О приложении", onClick: {
                viewModelWrapper.viewModel.navigateToAboutApp()
                action()
            })
        ]
        
        VStack {
            
            VStack {
                HStack {
                    Text("Дополнительно")
                        .font(.custom("Qanelas-Semibold", size: 20))
                        .foregroundColor(Color.black)
                    Spacer()
                }
                
                VStack {
                    ForEach(additionalList, id: \.id) { additional in
                        AdditionalItem(title: additional.title, onClick: additional.onClick)
                    }
                }
                .padding(.top, 20)
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 20)
        }
        .overlay(
               RoundedRectangle(cornerRadius: 5)
                .stroke(Color.Menu.Account.border, lineWidth: 1)
           )
    }
}

struct AdditionalItem : View {
    
    let title: String
    let onClick: () -> Void
    
    var body: some View {
        Button {
            onClick()
        } label: {
            HStack {
                Image("ic_square")
                    .resizable()
                    .foregroundColor(Color(hex: "A8ACAF"))
                    .frame(width: 18, height: 20)
                
                Text(title)
                    .font(.custom("Qanelas-Semibold", size: 16))
                    .foregroundColor(Color.black)
                    .padding(.leading, 18)
                
                Spacer()
                
                Image("ic_next")
                    .resizable()
                    .foregroundColor(Color(hex: "F73300"))
                    .frame(width: 5, height: 11)
            }
        }
    }
}

File: androidApp/src/main/java/com/mobile/fairless/android/features/menu/components/AdditionalView.kt
Ipackage com.mobile.fairless.android.features.menu.components

import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Icon
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.mobile.fairless.R
import com.mobile.fairless.android.di.StatefulViewModelWrapper
import com.mobile.fairless.android.theme.colors
import com.mobile.fairless.android.theme.fontQanelas
import com.mobile.fairless.features.menu.state.MenuState
import com.mobile.fairless.features.menu.viewModel.MenuViewModel

data class AdditionalInfo(
    val title: String,
    val onClick: () -> Unit
)

@Composable
fun AdditionalView(viewModelWrapper: StatefulViewModelWrapper<MenuViewModel, MenuState>) {
    val list = listOf(
        AdditionalInfo(title = stringResource(id = R.string.about), onClick = { viewModelWrapper.viewModel.navigateToAboutFairLess() }),
        AdditionalInfo(title = stringResource(id = R.string.rules), onClick = { viewModelWrapper.viewModel.navigateToRules() }),
        AdditionalInfo(title = stringResource(id = R.string.feedback), onClick = { viewModelWrapper.viewModel.navigateToFeedback() }),
        AdditionalInfo(title = "FAQ", onClick = { viewModelWrapper.viewModel.navigateToFaq() }),
        AdditionalInfo(title = stringResource(id = R.string.about_app), onClick = { viewModelWrapper.viewModel.navigateToAboutApp() }),
    )

    Column(
        Modifier
            .padding(top = 20.dp, start = 16.dp, end = 16.dp)
            .border(1.dp, colors.navBar, RoundedCornerShape(5.dp))
    ) {
        Column(
            Modifier
                .fillMaxWidth()
                .padding(top = 20.dp, bottom = 25.dp)
        ) {
            Text(
                text = "Дополнительно", style = TextStyle(
                    fontFamily = fontQanelas,
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 20.sp,
                    color = colors.black
                ), modifier = Modifier.padding(start = 16.dp, end = 16.dp, bottom = 20.dp)
            )
            list.forEach {
                AdditionalItem(additionalInfo = it)
            }
        }
    }
}

@Composable
fun AdditionalItem(additionalInfo: AdditionalInfo) {
    Row(
        Modifier
            .fillMaxWidth()
            .clickable {
                additionalInfo.onClick()
            },
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Row(
            modifier = Modifier.padding(start = 16.dp, top = 10.dp, bottom = 10.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                painter = painterResource(id = R.drawable.ic_square),
                contentDescription = "ic_square",
                modifier = Modifier.size(width = 18.dp, height = 20.dp),
                tint = Color(0xFFA8ACAF)
            )
            Text(
                text = additionalInfo.title, style = TextStyle(
                    fontFamily = fontQanelas,
                    fontWeight = FontWeight.SemiBold,
                    fontSize = 16.sp,
                    color = colors.black
                ), modifier = Modifier.padding(start = 18.dp)
            )
        }
        Row(
            Modifier
                .fillMaxWidth()
                .padding(end = 16.dp),
            horizontalArrangement = Arrangement.End
        ) {
            Icon(
                painter = painterResource(id = R.drawable.ic_next), contentDescription = "ic_next",
                modifier = Modifier.size(width = 5.dp, height = 11.dp),
                tint = colors.orangeMain
            )
        }
    }
}
